chips/cc26x0/src/uart.rs

```diff
@@ -185,10 +200,23 @@ impl UART {
         regs.dr.set(c as u32);
     }
 
+    pub fn read_byte(&self) -> u8 {
+        // Get byte from RX FIFO
+
+        while !self.rx_ready() {}
```

Comment:
The UART implementation can cause problems due to these "blocking-if-not-ready" in both RX and TX if the debugger isn't attached (if you're using the launchpad, you can test by removing the jumpers for UART between the board and the debugger), since it'll be blocking indefinitely.

We should probably add a way to detect whether or not UART is attached, and either not invoke any of these or simply return (eg. `if !self.initialized.get() { return }`)

Reply:
I'm going to rewrite the whole UART to look similar to the NRF52 interface, once I get DMA working, and I will make sure that this is included!